include/sys/dirent.h

```diff
@@ -66,7 +66,7 @@ struct dirent {
     off_t     d_off;              /**< \brief File offset */
     uint16_t  d_reclen;           /**< \brief Record length */
     uint8_t   d_type;             /**< \brief File type */
-    char      d_name[];           /**< \brief Filename */
+    char      d_name[NAME_MAX];   /**< \brief Filename */
```

Comment:
```suggestion
    char      d_name[0];   /**< \brief Filename */
```
Yes, I know, WTF.... This is a GNU extension from before FAMs being around which does not seem to freak the compiler out when its structure is used within another structure without being the last member in C++... Can you test that this works for you? I just tested locally with `scandir()`, but I'd feel better if you did another test. :)